### ISSUE TITLE
Redis Streams Consumer groups busted in Redis 5.0.3

### ISSUE THREAD

**Issue by NickCraver** [StackExchange/StackExchange.Redis#1027]
I'm not sure what happened here but 5.0.0-rc3 worked, 5.0.3 does not. There's a break somewhere I'm looking into now. I thought the build was flaky, but it turns out the Ubuntu AppVeyor image just updated Redis to latest.

Upon building from source locally for 5.0.3 on WSL, I see the same test failures that didn't exist on a 5.0.0-rc3 run just before. Logging here to track it, but going through release notes first ([link](https://raw.githubusercontent.com/antirez/redis/5.0/00-RELEASENOTES)) to see if there are any noted protocol changes.

Sample test output:
```
[xUnit.net 00:00:12.90]     PubSub.TestMassivePublishWithWithoutFlush_Local [SKIP]
Skipped  PubSub.TestMassivePublishWithWithoutFlush_Local
[xUnit.net 00:00:19.98]     Scripting.PurgeLuaScriptOnFinalize [SKIP]
Skipped  Scripting.PurgeLuaScriptOnFinalize
[xUnit.net 00:00:20.50]     Streams.StreamConsumerGroupReadFromStreamBeginning [FAIL]
[xUnit.net 00:00:20.52]     Streams.StreamDeleteConsumer [FAIL]
[xUnit.net 00:00:20.52]     Streams.StreamConsumerGroupAcknowledgeMessage [FAIL]
Failed   Streams.StreamConsumerGroupReadFromStreamBeginning
Error Message:
 Assert.True() Failure
Expected: True
Actual:   False
Stack Trace:
   at StackExchange.Redis.Tests.Streams.StreamConsumerGroupReadFromStreamBeginning() in C:\git\StackExchange\StackExchange.Redis\tests\StackExchange.Redis.Tests\Streams.cs:line 249
Standard Output Messages:
 Compiled under .NETCoreApp2.1
Running on: Microsoft Windows 10.0.17763
 01:47:27.773: 01:47:27.773: Service Counts: (Scheduler) Queue: 43436, Pool: 47, (Completion) Queue: 11906, Pool: 1157
 01:47:27.773: 01:47:27.773: Service Counts: (Scheduler) Queue: 43436, Pool: 47, (Completion) Queue: 11906, Pool: 1157


Failed   Streams.StreamDeleteConsumer
Error Message:
 Assert.Equal() Failure
Expected: 2
Actual:   0
Stack Trace:
   at StackExchange.Redis.Tests.Streams.StreamDeleteConsumer() in C:\git\StackExchange\StackExchange.Redis\tests\StackExchange.Redis.Tests\Streams.cs:line 796
Standard Output Messages:
 Compiled under .NETCoreApp2.1
Running on: Microsoft Windows 10.0.17763
 01:47:27.793: 01:47:27.793: Service Counts: (Scheduler) Queue: 43493, Pool: 47, (Completion) Queue: 11907, Pool: 1158
 01:47:27.793: 01:47:27.793: Service Counts: (Scheduler) Queue: 43493, Pool: 47, (Completion) Queue: 11907, Pool: 1158


Failed   Streams.StreamConsumerGroupAcknowledgeMessage
Error Message:
 Assert.True() Failure
Expected: True
Actual:   False
Stack Trace:
   at StackExchange.Redis.Tests.Streams.StreamConsumerGroupAcknowledgeMessage() in C:\git\StackExchange\StackExchange.Redis\tests\StackExchange.Redis.Tests\Streams.cs:line 318
Standard Output Messages:
 Compiled under .NETCoreApp2.1
Running on: Microsoft Windows 10.0.17763
 01:47:27.800: 01:47:27.800: Service Counts: (Scheduler) Queue: 43520, Pool: 47, (Completion) Queue: 11907, Pool: 1158
 01:47:27.800: 01:47:27.800: Service Counts: (Scheduler) Queue: 43520, Pool: 47, (Completion) Queue: 11907, Pool: 1158


[xUnit.net 00:00:20.59]     Streams.StreamConsumerGroupViewPendingInfoSummary [FAIL]
Failed   Streams.StreamConsumerGroupViewPendingInfoSummary
Error Message:
 Assert.True() Failure
Expected: True
Actual:   False
Stack Trace:
   at StackExchange.Redis.Tests.Streams.StreamConsumerGroupViewPendingInfoSummary() in C:\git\StackExchange\StackExchange.Redis\tests\StackExchange.Redis.Tests\Streams.cs:line 667
Standard Output Messages:
 Compiled under .NETCoreApp2.1
Running on: Microsoft Windows 10.0.17763
 01:47:27.872: 01:47:27.872: Service Counts: (Scheduler) Queue: 43694, Pool: 47, (Completion) Queue: 11907, Pool: 1158
 01:47:27.872: 01:47:27.872: Service Counts: (Scheduler) Queue: 43694, Pool: 47, (Completion) Queue: 11907, Pool: 1158


[xUnit.net 00:00:20.65]     Streams.StreamConsumerGroupClaimMessagesReturningIds [FAIL]
Failed   Streams.StreamConsumerGroupClaimMessagesReturningIds
Error Message:
 Assert.Equal() Failure
Expected: 3
Actual:   4
Stack Trace:
   at StackExchange.Redis.Tests.Streams.StreamConsumerGroupClaimMessagesReturningIds() in C:\git\StackExchange\StackExchange.Redis\tests\StackExchange.Redis.Tests\Streams.cs:line 419
Standard Output Messages:
 Compiled under .NETCoreApp2.1
Running on: Microsoft Windows 10.0.17763
 01:47:27.924: 01:47:27.924: Service Counts: (Scheduler) Queue: 43853, Pool: 47, (Completion) Queue: 11908, Pool: 1158
 01:47:27.924: 01:47:27.924: Service Counts: (Scheduler) Queue: 43853, Pool: 47, (Completion) Queue: 11908, Pool: 1158


[xUnit.net 00:00:20.67]     Streams.StreamConsumerGroupReadMultipleRestrictCount [FAIL]
Failed   Streams.StreamConsumerGroupReadMultipleRestrictCount
Error Message:
 Assert.True() Failure
Expected: True
Actual:   False
Stack Trace:
   at StackExchange.Redis.Tests.Streams.StreamConsumerGroupReadMultipleRestrictCount() in C:\git\StackExchange\StackExchange.Redis\tests\StackExchange.Redis.Tests\Streams.cs:line 588
Standard Output Messages:
 Compiled under .NETCoreApp2.1
Running on: Microsoft Windows 10.0.17763
 01:47:27.948: 01:47:27.948: Service Counts: (Scheduler) Queue: 43924, Pool: 47, (Completion) Queue: 11908, Pool: 1158
 01:47:27.948: 01:47:27.948: Service Counts: (Scheduler) Queue: 43924, Pool: 47, (Completion) Queue: 11908, Pool: 1158


[xUnit.net 00:00:20.75]     Streams.StreamConsumerGroupReadMultipleOneReadBeginningOneReadNew [FAIL]
Failed   Streams.StreamConsumerGroupReadMultipleOneReadBeginningOneReadNew
Error Message:
 Assert.True() Failure
Expected: True
Actual:   False
Stack Trace:
   at StackExchange.Redis.Tests.Streams.StreamConsumerGroupReadMultipleOneReadBeginningOneReadNew() in C:\git\StackExchange\StackExchange.Redis\tests\StackExchange.Redis.Tests\Streams.cs:line 469
Standard Output Messages:
 Compiled under .NETCoreApp2.1
Running on: Microsoft Windows 10.0.17763
 01:47:28.032: 01:47:28.032: Service Counts: (Scheduler) Queue: 44123, Pool: 47, (Completion) Queue: 11908, Pool: 1158
 01:47:28.032: 01:47:28.032: Service Counts: (Scheduler) Queue: 44123, Pool: 47, (Completion) Queue: 11908, Pool: 1158
```

**Comment by ttingen** [comment]
It looks like the change was made in 5.0.2.

When the consumer group initially reads from the stream, some tests are passing "0-0" as the starting point for the read. That appears to no longer be valid for the *initial read*. Once new messages have been read into the consumer group, you can pass "0-0" and get results.

I updated the `StreamConsumerGroupReadFromStreamBeginning` test's StreamReadGroup call to send StreamPosition.NewMessages instead of "0-0" for the initial read and that fixed the test.

The same change also fixed  the `StreamConsumerGroupAcknowledgeMessage` test.

I didn't get a chance to look at the failing tests for reading multiple consumer groups, I'll look at those tomorrow.

**Comment by NickCraver** [comment]
Fixed in #1028
